src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -629,18 +648,18 @@ private async Task EnqueueWorkItemForSolutionAsync(Solution solution, Invocation
 
             private async Task EnqueueWorkItemAfterDiffAsync(Solution oldSolution, Solution newSolution, ProjectId projectId)
             {
-                var oldProject = oldSolution.GetProject(projectId);
-                var newProject = newSolution.GetProject(projectId);
+                var oldProject = oldSolution.GetRequiredProject(projectId);
+                var newProject = newSolution.GetRequiredProject(projectId);
 
                 await EnqueueWorkItemAsync(newProject.GetChanges(oldProject)).ConfigureAwait(continueOnCapturedContext: false);
             }
 
             private async Task EnqueueWorkItemAfterDiffAsync(Solution oldSolution, Solution newSolution, DocumentId documentId)
             {
-                var oldProject = oldSolution.GetProject(documentId.ProjectId);
-                var newProject = newSolution.GetProject(documentId.ProjectId);
+                var oldProject = oldSolution.GetRequiredProject(documentId.ProjectId);
+                var newProject = newSolution.GetRequiredProject(documentId.ProjectId);
```

Comment:
No worry that this gets called during added/removed operations and the nulls were allowed before?

Reply:
It would have thrown NRE if the projects didn't exist.

Reply:
The docs are also required by EnqueueWorkItemAsync .